db/db.go

```diff
@@ -2267,41 +2269,46 @@ func GetWithdrawals(query string, length, start uint64, orderBy, orderDir string
 		orderBy = "block_slot"
 	}
 
-	if query != "" {
-		bquery, _ := hex.DecodeString(strings.TrimPrefix(query, "0x"))
+	withdrawalsQuery := `
+		SELECT 
+			w.block_slot as slot,
+			w.withdrawalindex as index,
+			w.validatorindex,
+			w.address,
+			w.amount
+		FROM blocks_withdrawals w
+		INNER JOIN blocks b ON w.block_root = b.blockroot AND b.status = '1'
+		%s
+		ORDER BY %s %s
+		LIMIT $1
+		OFFSET $2`
+
+	trimmedQuery := strings.ToLower(strings.TrimPrefix(query, "0x"))
+	if trimmedQuery != "" {
+		searchQuery := ""
+		if uiQuery, err := strconv.ParseUint(query, 10, 64); err == nil {
+			searchQuery = `
+				WHERE w.validatorindex = $3
+					OR block_slot = $3
+					OR (block_slot / $5) = $3
+					OR ENCODE(address, 'hex') LIKE ($4 || '%')`
+
+			err := ReaderDb.Select(&withdrawals, fmt.Sprintf(withdrawalsQuery, searchQuery, orderBy, orderDir),
+				length, start, uiQuery, trimmedQuery, utils.Config.Chain.Config.SlotsPerEpoch)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
**Minor:** You should "centralize" the error handling by moving it down below the if starting at line `2287`. No need for it to be part of every possible if path when there is no unique error handling based on the path.

This also applies to `func GetBLSChanges`: https://github.com/gobitfly/eth2-beaconchain-explorer/blob/9e87cca798e6d7d590ab4b459d9f5acbe1db6842/db/db.go#L2874

Reply:
I have added a commit to centralize this a bit more. ptal